examples/no_std/rust-toolchain

```diff
@@ -0,0 +1 @@
+nightly
```

Comment:
The no_std application (more specifically, the cortex-m crates) need a newer Rust version than the one that's pinned in the repo root's `rust-toolchain`. IIUC it needs to be newer than 1.59.
I guess here we'd need to pin a version, too, because just `nightly` is a moving target.

Any suggestions what to pin? Should I just use the first one that is new enough to build?

Reply:
Let's try to move everything to the most recent nightly. We should update anyway. In the best case it just works. Can you give that a try?

Reply:
yeah, that's not gonna fly, the cortex-m crates depend on stable `asm!()`.

But, IIUC, the hacspec-lib rust code compiles with newer nightlies. `language/` doesn't, but is not needed to actually execute hacspec code? In that case, it should be fine to pin only this application to a newer nightly?

Reply:
Yeah, pulling the test out would speed things up. Maybe just move it into a follow up?
To ensure that no_std is working we can change the CI to just build the library for a no_std target for now.